core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -319,6 +381,556 @@ public void testNamespaceWithDot() {
     Assert.assertFalse("Namespace should not exist", catalog.namespaceExists(withDot));
   }
 
+  @Test
```

Comment:
nit: would it make sense to pull these out into a separate PR so that we get those in sooner? Unless this PR won't take too long to be merged of course